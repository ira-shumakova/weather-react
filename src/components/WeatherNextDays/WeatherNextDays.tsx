import { Grid } from "@mui/material";
import { IDaysWeather } from "../../dto/types";
import WeatherNextOneDay from "../WeatherNextOneDay/WeatherNextOneDay";

type Props = {
    weather: IDaysWeather[],
}

const WeatherNextDays = ({weather}: Props) => {
  return (
      <Grid
        container spacing={2}
        justifyContent="space-around"
        alignItems="center"

      >
        {weather.slice(1,7).map((day, index) => 
          <Grid item xs={2} key={index}>
            <WeatherNextOneDay day={day} />
          </Grid>
        )}
      </Grid>
  )
}

export default WeatherNextDays;