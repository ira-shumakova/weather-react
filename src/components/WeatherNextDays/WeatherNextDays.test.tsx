import { render } from "@testing-library/react";
import WeatherNextDays from "./WeatherNextDays";

describe("WeatherNextDays", () => {
  test("matches snapshot", () => {
    const weather = [
      {
        temp: {
          max: 14,
          min: 6,
        },
        feels_like: 13,
        humidity: 55,
        pressure: 1017,
        sunrise: 1681960920,
        sunset: 1682011401,
        wind_speed: 6.68,
        dt: 1681984800,
        weather: [
          {
            description: "broken clouds",
            icon: "04d",
            main: "Clouds"
          },
        ],
      },
    ];
    const {container} = render(<WeatherNextDays weather={weather} />)
    expect(container).toMatchSnapshot();
    });
});