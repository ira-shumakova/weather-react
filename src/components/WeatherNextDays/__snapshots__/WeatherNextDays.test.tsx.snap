// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherNextDays matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1hx6p0u-MuiGrid-root"
  />
</div>
`;
